src/EncryptionOracle.sol

```diff
@@ -196,13 +156,50 @@ abstract contract EncryptionOracle is ThresholdNetwork, IEncryptionOracle, Ownab
         } catch Error(string memory reason) {
             revert OracleResultFailed(reason);
         } catch {
-            revert OracleResultFailed("Client does not support oracleResult() method");
+            revert OracleResultFailed(
+                "Client does not support oracleResult() method"
+            );
+        }
+    }
+
+    function estimateDeliverReencryption(
+        uint256 _requestId,
+        ReencryptedCipher calldata _cipher,
+        address callbackRecipient
+    ) external whenNotPaused onlyRelayer returns (bool) {
+        PendingRequest memory pr = pendingRequests[_requestId];
+        if (pr.client != address(0)) {
+            delete pendingRequests[_requestId];
+        } else {
+            pr.client = callbackRecipient;
+            pr.gasReimbursement = 1;
+        }
+        IEncryptionClient client = IEncryptionClient(pr.client);
```

Comment:
Also seems to me that from this line it's the same as in the original function ? Can you refactor and put the logic in one place ?